proto/cosmos/orm/v1/orm.proto

```diff
@@ -0,0 +1,114 @@
+syntax = "proto3";
+
+package cosmos.orm.v1alpha1;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/orm/types/ormpb";
+
+extend google.protobuf.MessageOptions {
+
+  // table specifies that this message will be used as an ORM table. It cannot
+  // be used together with the singleton option.
+  TableDescriptor table = 100003790;
+
+  // singleton specifies that this message will be used as an ORM singleton. It cannot
+  // be used together with the table option.
+  SingletonDescriptor singleton = 100003791;
+}
+
+// TableDescriptor describes an ORM table.
+message TableDescriptor {
+
+  // primary_key defines the primary key for the table.
+  PrimaryKeyDescriptor primary_key = 1;
+
+  // index defines one or more secondary indexes.
+  repeated SecondaryIndexDescriptor index = 2;
+
+  // id is a non-zero integer ID that must be unique within the
+  // tables and singletons in this file. It may be deprecated in the future when this
+  // can be auto-generated.
+  uint32 id = 3;
+}
+
+// PrimaryKeyDescriptor describes a table primary key.
+message PrimaryKeyDescriptor {
+
+  // fields is a comma-separated list of fields in the primary key. Spaces are
```

Comment:
How about `/` or `:` separated fields?

Which looks best `foo/bar`, `foo:bar` or `foo,bar`?

Reply:
I assume `foo, bar` is more canonical.

Reply:
allowing spaces? that's one thing which is confusing about `,` - both `foo,bar` and `foo, bar` could be seen as correct. whereas `foo/ bar` is obviously wrong. however `/` denotes hierarchy and this isn't really "hierarchical"